streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -703,6 +703,14 @@ void runOnce() {
             // we record the ratio out of the while loop so that the accumulated latency spans over
             // multiple iterations with reasonably large max.num.records and hence is less vulnerable to outliers
             taskManager.recordTaskProcessRatio(totalProcessLatency, now);
+
+            // Only log this when we actually processed some records to avoid flooding INFO-level logs when there are
+            // no records on the input topics
+            if (totalProcessed > 0) {
+                log.info("Finished processing {} records, going to poll again", totalProcessed);
+            } else {
+                log.debug("Processed zero records, going to poll again");
```

Comment:
Is it important to make the distinction between "Processed zero records, going to poll again" and "Finished processing 0 records, going to poll again"?

Also if we are not saying its processed anything maybe we should log what it is pulling from

Reply:
Reworked it a bit, lmk what you think

Reply:
Looks good